src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -584,24 +584,27 @@
       }
    }
    
-   # if the object exists in the cache environment, return it. objects
-   # in the cache environment have already been coerced to data frames.
-   if (exists(cacheKey, where = .rs.CachedDataEnv, inherits = FALSE))
-      return(get(cacheKey, envir = .rs.CachedDataEnv, inherits = FALSE))
-   
-   # perhaps the object has been saved? attempt to load it into the
-   # cached environment
-   cacheFile <- file.path(cacheDir, paste(cacheKey, "Rdata", sep = "."))
-   if (file.exists(cacheFile))
+   if (.rs.isNonEmptyScalarString(cacheKey))
    {
-      status <- try(load(cacheFile, envir = .rs.CachedDataEnv), silent = TRUE)
-      if (inherits(status, "try-error"))
-         return(NULL)
-      
+      # if the object exists in the cache environment, return it. objects
+      # in the cache environment have already been coerced to data frames.
       if (exists(cacheKey, where = .rs.CachedDataEnv, inherits = FALSE))
          return(get(cacheKey, envir = .rs.CachedDataEnv, inherits = FALSE))
+      
+      # perhaps the object has been saved? attempt to load it into the
+      # cached environment
+      cacheFile <- file.path(cacheDir, paste(cacheKey, "Rdata", sep = "."))
+      if (file.exists(cacheFile))
+      {
+         status <- try(load(cacheFile, envir = .rs.CachedDataEnv), silent = TRUE)
+         if (inherits(status, "try-error"))
+            return(NULL)
+         
+         if (exists(cacheKey, where = .rs.CachedDataEnv, inherits = FALSE))
+            return(get(cacheKey, envir = .rs.CachedDataEnv, inherits = FALSE))
+      }
    }
-   
+
```

Comment:
this diff is confusing -- the existing code was just wrapped with `if (.rs.isNonEmptyScalarString(cacheKey))`

Reply:
This might be easier to read (left = `main`, right = this PR)

<img width="1247" alt="image" src="https://github.com/rstudio/rstudio/assets/25834218/1cdb16fe-ca2e-42f9-b428-c3bc90b6d9e5">

Reply:
For what it's worth, the diff renders clearly for me if I use "hide whitespace" in the file view, e.g. https://github.com/rstudio/rstudio/pull/13238/files?w=1. 

<img width="664" alt="Screenshot 2023-06-14 at 10 35 41 AM" src="https://github.com/rstudio/rstudio/assets/1976582/29e64523-781c-40df-96f8-13594b76bc25">

Reply:
ah yes that's so much better 🙌